src/BucketTesting/Domain/Model/BucketLog.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\BucketTesting\Domain\Model;
+
+use DateTime;
+use DateTimeInterface;
+
+class BucketLog {
+
+	private int $id;
+	private int $external_id;
+	private string $eventName;
+	/** @var BucketLogBucket[] */
+	private array $buckets;
+	private DateTimeInterface $date;
+
+	public function __construct( int $external_id, string $eventName, ?DateTime $date = null ) {
+		$this->external_id = $external_id;
+		$this->eventName = $eventName;
+		$this->date = $date ?? new DateTime();
+	}
+
+	public function getId(): int {
+		return $this->id;
+	}
+
+	public function getExternalId(): int {
+		return $this->external_id;
+	}
+
+	public function getDate(): DateTimeInterface {
+		return $this->date;
+	}
+
+	public function getEventName(): ?string {
+		return $this->eventName;
+	}
+
+	/**
+	 * @return BucketLogBucket[]
+	 */
+	public function getBuckets(): array {
+		return $this->buckets;
+	}
+
+	/**
+	 * @param BucketLogBucket[] $buckets
+	 */
+	public function setBuckets( array $buckets ): void {
+		$this->buckets = $buckets;
+	}
```

Comment:
Internally, the Buckets should be an `ArrayCollection`, see https://www.doctrine-project.org/projects/doctrine-orm/en/2.7/reference/association-mapping.html and the setter should pass the object(s)/array on to the collection. Or use an `addBucket` method instead of a setter.